### ISSUE TITLE
The constraint options don't seem to be implemented any more

### ISSUE THREAD

**Issue by zogtrog** [sebastienros/jint#799]
The example code does not work anymore for constraints

            var engine = new Jint.Engine(options =>
            {

                // Limit memory allocations to MB
                options.LimitMemory(4_000_000);

                // Set a timeout to 4 seconds.
                options.TimeoutInterval(TimeSpan.FromSeconds(4));

                // Set limit of 1000 executed statements.
                options.MaxStatements(1000);

                // Use a cancellation token.
                options.CancellationToken(cancellationToken);
            });

ERRORS IN VS2017

Severity	Code	Description	Project	File	Line	Suppression State
Error	CS1061	'Options' does not contain a definition for 'CancellationToken' and no accessible extension method 'CancellationToken' accepting a first argument of type 'Options' could be found (are you missing a using directive or an assembly reference?)	TestScripting	
Error	CS1061	'Options' does not contain a definition for 'LimitMemory' and no accessible extension method 'LimitMemory' accepting a first argument of type 'Options' could be found (are you missing a using directive or an assembly reference?)	TestScripting	
Error	CS1061	'Options' does not contain a definition for 'MaxStatements' and no accessible extension method 'MaxStatements' accepting a first argument of type 'Options' could be found (are you missing a using directive or an assembly reference?)	
Error	CS1061	'Options' does not contain a definition for 'TimeoutInterval' and no accessible extension method 'TimeoutInterval' accepting a first argument of type 'Options' could be found (are you missing a using directive or an assembly reference?)

**Comment by sebastienros** [comment]
Everything seems fine:

https://github.com/sebastienros/jint/blob/dev/Jint/Constraints/ConstraintsOptionsExtensions.cs

And unit tests still work. Are you using 3.0 versions?

**Comment by zogtrog** [comment]
I used NuGet to install JInt into VS2017. I was using the latest version 3.0.0-beta-1914. I tried downgrading to 2.11.58.
I need to use constraints in JINT because JINT could be running code supplied by third parties which might be hostile or have memory bombs in it. I want to limit recursions and the amount of memory available for a scripting task. 
Which version should I be using ?

**Comment by lahma** [comment]
Please use the latest beta on NuGet. If you have `using Jint` on top of your file the extension methods should kick in.

**Comment by zogtrog** [comment]
Cheerd lahms - that was it for some reason the code completer was giving a list of incomplete options. I have ensured I am using the latest beta. Putting using Jint right at the top solved the problem. 

I had just using Jint.Engine which gave me sufficient options to look like I had included the all the necessary assemblies 

Sorry sebastienros for taking your time. 

I am a goober.

This now works.

 var JSE = new Jint.Engine(options =>
             {
                 options.LimitRecursion(10);
                 options.LimitMemory(max_mem);
             });
